src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+            # polygon with our base polygon at the correct z position, and then output the resulting polygon.
+            return PolygonalRegion(polygon=self.polygon, z=other.z).intersect(other)
+
+        return super().intersect(other, triedReversed)
+
+    def union(self, other, triedReversed=False):
+        """ Get a `Region` representing the union of this region with another.
+
+        This function handles union computation for `PolygonalFootprintRegion` with:
+        * `PolygonalFootprintRegion`
+        """
+        if isinstance(other, PolygonalFootprintRegion):
+            # Other region is a PolygonalFootprintRegion, so we can just union the base polygons
+            # and take the footprint of the result.
+            return PolygonalRegion(polygon=self.polygon).union(PolygonalRegion(polygon=other.polygon)).footprint
+
+        return super().union(other, triedReversed)
+
+    def difference(self, other):
+        """ Get a `Region` representing the difference of this region with another.
+
+        This function handles difference computation for `PolygonalFootprintRegion` with:
+        * `PolygonalFootprintRegion`
+        """
+        if isinstance(other, PolygonalFootprintRegion):
+            # Other region is a PolygonalFootprintRegion, so we can just difference the base polygons
+            # and take the footprint of the result, if it isn't empty.
+            new_poly = PolygonalRegion(polygon=self.polygon).difference(PolygonalRegion(polygon=other.polygon))
+
+            if isinstance(new_poly, EmptyRegion):
+                return new_poly
+            else:
+                return new_poly.footprint
+
+        return super().difference(other)
+
+    def uniformPointInner(self):
+        raise UndefinedSamplingException(f'Attempted to sample from a PolygonalFootprintRegion, for which uniform sampling is undefined')
+
+    def containsPoint(self, point):
+        """ Checks if a point is contained in the polygonal footprint.
+
+        Equivalent to checking if the (x, y) values are contained in the polygon.
+
+        Args:
+            point: A point to be checked for containment.
+        """
+        x_y_point = toVector(point)[:2]
+        return self.polygon.intersects(shapely.geometry.Point(x_y_point))
+
+    def containsObject(self, obj):
+        """ Checks if an object is contained in the polygonal footprint.
+
+        Args:
+            obj: An object to be checked for containment.
+        """
+        # Check containment using the bounding polygon of the object.
+        return self.polygon.contains(obj._boundingPolygon)
+
+    def containsRegionInner(self, reg, tolerance):
+        buffered_polygons = self.polygons.buffer(tolerance)
+
+        if isinstance(other, MeshRegion):
+            return buffered_polygons.contains(reg._boundingPolygon)
+
+        if isinstance(other, (PolygonalRegion, PolygonalFootprintRegion)):
+            return buffered_polygons.contains(reg.poylgons)
```

Comment:
```suggestion
            return buffered_polygons.contains(reg.polygons)
```
No tests for these cases I guess...